internal/dlog/dlog_test.go

```diff
@@ -30,19 +31,21 @@ type params struct {
 	p, order, g *big.Int
 }
 
-func get_params() *params {
-	p, _ := emmy.GetSafePrime(20)
-	pMin1 := new(big.Int).Sub(p, big.NewInt(1))
-	x := emmy.GetRandomInt(p)
+func get_params(t *testing.T) *params {
+	key, err := keygen.NewElGamal(20)
+	if err != nil {
+		t.Fatalf("Error during parameters generation: %v", err)
+	}
+	pMin1 := new(big.Int).Sub(key.P, big.NewInt(1))
 	return &params{
-		p:     p,
-		order: new(big.Int).Div(pMin1, big.NewInt(2)),
-		g:     new(big.Int).Exp(x, big.NewInt(2), p),
+		p:     key.P,
+		order: pMin1,
+		g:     key.G,
 	}
 }
 
 func TestDLog(t *testing.T) {
-	params := get_params()
+	params := get_params(t)
```

Comment:
See comment above. It would be better to have it like this:
````
params, err := get_params()
if err != nil {
    t.Fatalf("Error during parameters generation: %v", err)
}
````